test/common/docs.js

```diff
@@ -0,0 +1,94 @@
+/**
+ * Copyright 2014 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/*global describe, it */
+
+'use strict';
+
+var vm = require('vm');
+var assert = require('assert');
+var gcloud = require('../../');
+var glob = require('glob');
+var mitm = require('mitm');
+
+var sandbox = {
+  gcloud: gcloud,
+  require: require,
+  process: process,
+  Buffer: Buffer
+};
+
+function runCodeInSandbox(code) {
+  vm.createContext(sandbox);
+  try {
+    vm.runInNewContext(code, sandbox, {
+      filename: 'assert-code.vm',
+      timeout: 1000
+    });
+  } catch(err) {
+    // rethrow the error with code for context and resolving issues faster.
+    throw new Error('\n' + code + '\n\n' + err.message);
+  }
+}
+
+describe('docs', function() {
+  var MITM;
+  var FILES;
+
+  before(function(done) {
+    // Turn off the network so that API calls aren't actually made.
+    MITM = mitm();
+    glob('docs/json/**/*.json', {}, function(err, files) {
+      assert.ifError(err);
+      FILES = files;
+      done();
+    });
+  });
+
+  after(function() {
+    // Re-enable the network.
+    MITM.disable();
+  });
+
+  it('should run public docs examples without errors', function() {
+    for (var f in FILES) {
+      var fileDocBlocks = require('../../' + FILES[f]);
+      for (var i in fileDocBlocks) {
+        var doc = fileDocBlocks[i];
+
+        // We don't want to test developer docs or private documentation that
+        // are hidden because they use the modules in a less-recommended way
+        // e.g. use class object directly or call obj.makeReq() directly.
+        if (!doc.isPrivate) {
```

Comment:
I am working to remove this restriction so all the docs are tested! :) The more, the better! Death to bitrot!